stdlib/src/os/env.mojo

```diff
@@ -51,6 +51,26 @@ fn setenv(name: String, value: String, overwrite: Bool = True) -> Bool:
     return status == 0
 
 
+fn unsetenv(name: String) -> Bool:
+    """Unsets an environment variable.
+
+    Constraints:
+        The function only works on macOS or Linux and returns False otherwise.
```

Comment:
**Question** Any particular reason not to just use `constrained` instead (hard error in other words) instead of just gracefully returning `False` on Windows?

Reply:
No concerns here, I was following the same pattern as `getenv` and `setenv`. I'll open up a separate PR to make them consistent with using `constrained` and `c_int`. For now I've updated only `unsetenv`.